src/parser.rs

```diff
@@ -884,6 +942,15 @@ impl Get for Attributes {
     }
 }
 
+fn read_text(parser: &mut Reader) -> Result<Option<String>, Error> {
+    let ret = match try!(parser.next()) {
+        Characters(text) => Ok(Some(text)),
+        _ => Ok(None),
+    };
+    try!(ignore_element(parser));
+    ret
+}
+
```

Comment:
Like I said previously I would like this function to resemble other `read_*` functions i.e. have a loop which handles `EndElement` explicitly, concatenates multiple `Characters` events if necessary and returns an error on `StartElement`.